src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift

```diff
@@ -0,0 +1,475 @@
+		PPAnalyticsCollector.submitter = submitter
+	}
+	
+	#if !RELEASE
+
+	/// Setup Analytics for testing. The store or the submitter can be nil for testing purposes.
+	static func setupMock(
+		store: (Store & PPAnalyticsData)? = nil,
+		submitter: PPAnalyticsSubmitter? = nil
+	) {
+		PPAnalyticsCollector.store = store
+		PPAnalyticsCollector.submitter = submitter
+	}
+	
+	#endif
+
+	/// The main purpose for the collector. Call this method to log some analytics data and pass the corresponding enums.
+	static func collect(_ dataType: PPADataType) {
+
+		// Make sure the user consent is given. If not, we must not log something.
+		guard let consent = store?.isPrivacyPreservingAnalyticsConsentGiven,
+			  consent == true else {
+			Log.info("Forbidden to log any analytics data due to missing user consent", log: .ppa)
+			return
+		}
+
+		Log.debug("Logging analytics data: \(dataType)", log: .ppa)
+		switch dataType {
+		case let .userData(userMetadata):
+			Analytics.logUserMetadata(userMetadata)
+		case let .riskExposureMetadata(riskExposureMetadata):
+			Analytics.logRiskExposureMetadata(riskExposureMetadata)
+		case let .clientMetadata(clientMetadata):
+			Analytics.logClientMetadata(clientMetadata)
+		case let .testResultMetadata(TestResultMetadata):
+			Analytics.logTestResultMetadata(TestResultMetadata)
+		case let .keySubmissionMetadata(keySubmissionMetadata):
+			Analytics.logKeySubmissionMetadata(keySubmissionMetadata)
+		case let .exposureWindowsMetadata(exposureWindowsMetadata):
+			Analytics.logExposureWindowsMetadata(exposureWindowsMetadata)
+		case let .submissionMetadata(submissionMetadata):
+			Analytics.logSubmissionMetadata(submissionMetadata)
+		}
+
+		// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
+		Analytics.triggerAnalyticsSubmission()
+	}
+
+	/// This removes all stored analytics data that we collected.
+	static func deleteAnalyticsData() {
+		store?.currentRiskExposureMetadata = nil
+		store?.previousRiskExposureMetadata = nil
+		store?.userMetadata = nil
+		store?.lastSubmittedPPAData = nil
+		store?.lastAppReset = nil
+		store?.lastSubmissionAnalytics = nil
+		store?.clientMetadata = nil
+		store?.testResultMetadata = nil
+		store?.keySubmissionMetadata = nil
+		store?.exposureWindowsMetadata = nil
+		Log.info("Deleted all analytics data in the store", log: .ppa)
+	}
+
+	/// Triggers the submission of all collected analytics data. Only if all checks success, the submission is done. Otherwise, the submission is aborted. Optionally, you can specify a completion handler to get the success or failure handlers.
+	static func triggerAnalyticsSubmission(completion: ((Result<Void, PPASError>) -> Void)? = nil) {
+		guard let submitter = submitter else {
+			Log.warning("I cannot submit analytics data. Perhaps i am a mock or setup was not called correctly?", log: .ppa)
+			return
+		}
+		submitter.triggerSubmitData(completion: completion)
+	}
+
+
+	#if !RELEASE
```

Comment:
I would consider to move all this !RELEASE code out into a separate extension and maybe even another file (`PPAnalyticsCollector+Testing.swift`). This reduces file size and separates production from dev code.

Reply:
Good idea. I will have a try.

Reply:
Move it into one extension to separate it from production code goes easy and is good idea.
Move it into a separate file need to make private properties internal and this I don't like. So i will only move it into an extension.